File: sample/src/main/java/tech/thdev/androidrecyclerview/data/source/image/ImagesMetaDataSource.java
package tech.thdev.androidrecyclerview.data.source.image;

import java.util.List;

import io.reactivex.Observable;
import tech.thdev.androidrecyclerview.data.LocalImage;

/**
 * Created by Tae-hwan on 18/10/2016.
 */

public interface ImagesMetaDataSource {

    Observable<List<LocalImage>> getAllImages();
}

File: sample/src/main/java/tech/thdev/androidrecyclerview/data/source/image/ImagesMetaDataSource.kt
package tech.thdev.androidrecyclerview.data.source.image

import io.reactivex.rxjava3.core.Flowable
import tech.thdev.androidrecyclerview.data.LocalImage

/**
 * Created by Tae-hwan on 18/10/2016.
 */
interface ImagesMetaDataSource {
    fun getAllImages(): Flowable<List<LocalImage>>
}